staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/vault/client.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package vault implements envelop encryption provider based on Vault KMS
+package vault
+
+import (
+	"fmt"
+	"path"
+	"reflect"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// Handle all communication with Vault server.
+type clientWrapper struct {
+	client      *api.Client
+	encryptPath string
+	decryptPath string
+	authPath    string
+}
+
+// Initialize a client wrapper for vault kms provider.
+func newClientWrapper(config *EnvelopeConfig) (*clientWrapper, error) {
+	client, err := newVaultClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Vault transit path is configurable. "path", "/path", "path/" and "/path/"
+	// are the same.
+	transit := "transit"
+	if config.TransitPath != "" {
+		transit = config.TransitPath
+	}
+
+	// auth path is configurable. "path", "/path", "path/" and "/path/" are the same.
+	auth := "auth"
+	if config.AuthPath != "" {
+		auth = config.AuthPath
+	}
+	wrapper := &clientWrapper{
+		client:      client,
+		encryptPath: path.Join("v1", transit, "encrypt"),
+		decryptPath: path.Join("v1", transit, "decrypt"),
+		authPath:    path.Join(auth),
+	}
+
+	// Set token for the api.client.
+	if config.Token != "" {
```

Comment:
Stylistically for the project, more like
```go
if len(config.Token) != 0{
   client.SetToken
   return wrapper, nil
}

if err := wrapper.refreshToken(config); err != nil{
   return nil, err
}
return wrapper, nil
```

Reply:
There is logical difference between the two, I am keeping the original.